qiskit_nature/transformers/freeze_core_transformer.py

```diff
@@ -0,0 +1,110 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Freeze-Core Reduction interface."""
+
+from typing import List, Optional
+import logging
+
+from .active_space_transformer import ActiveSpaceTransformer
+from ..drivers import QMolecule
+
+logger = logging.getLogger(__name__)
+
+
+class FreezeCoreTransformer(ActiveSpaceTransformer):
+    """The Freeze-Core reduction."""
+
+    def __init__(self, freeze_core: bool = True,
+                 remove_orbitals: Optional[List[int]] = None,
+                 ):
+        """Initializes a transformer which reduces a `QMolecule` by removing some molecular
+        orbitals.
+
+        The orbitals to be removed are specified in two ways:
+            1. When `freeeze_core` is enabled (the default), the `core_orbitals` listed in the
+               `QMolecule` are made inactive and removed in the same fashion as in the
+               :class:`ActiveSpaceTransformer`.
+            2. Additionally, unoccupied molecular orbitals can be removed via a list of indices
+               passed to `remove_orbitals`. It is the user's responsibility to ensure that these are
+               indeed unoccupied orbitals, as no checks are performed.
+
+        If you want to remove additional occupied orbitals, please use the
+        :class:`ActiveSpaceTransformer` instead.
+
+        Args:
+            freeze_core: A boolean indicating whether to remove the molecular orbitals specified by
+                        `QMolecule.core_orbitals.
+            remove_orbitals: A list of indices specifying molecular orbitals which are removed.
+                             No checks are performed on the nature of these orbitals, so the user
+                             must make sure that these are _unoccupied_ orbitals, which can be
+                             removed without taking any energy shifts into account.
+        """
+        self._freeze_core = freeze_core
+        self._remove_orbitals = remove_orbitals
+
+        super().__init__()
+
+    def transform(self, q_molecule: QMolecule) -> QMolecule:
+        """Reduces the given `QMolecule` by removing the core and optionally defined unoccupied
+        molecular orbitals.
+
+        Args:
+            q_molecule: the `QMolecule` to be transformed.
+
+        Returns:
+            A new `QMolecule` instance.
+
+        Raises:
+            QiskitNatureError: If more electrons or orbitals are requested than are available, if an
+                               uneven number of inactive electrons remains, or if the number of
+                               selected active orbital indices does not match
+                               `num_molecular_orbitals`.
+        """
+        q_molecule_new = super().transform(q_molecule)
```

Comment:
Ah ok, I'll look more over these later. I had thought perhaps that you would create an ActiveSpaceTransformer at this point, given that the QMolecule is now known, and just figure the correct active space configuration and just call its transform. This would then just extend the base Transformer class and delegate to ActiveSpaceTransformer which would just be created/used internally. Maybe the interface did not permit the removal in the way that this does hence the sub-classing. Though the dictionary keys then for the energy extracted might not be as expected unless this was documented as a convenience that uses ActiveSpaceTransformer to simply set a frozen core config

Reply:
That would be an option to but I found this more intuitive. Since this does not change anything user-facing we can change this later if we want to.